packages/contracts/src/EmailAccountRecovery.sol

```diff
@@ -185,14 +223,17 @@ abstract contract EmailAccountRecovery {
             emailAuthImplementation(),
             abi.encodeCall(
                 EmailAuth.initialize,
-                (address(this), emailAuthMsg.proof.accountSalt)
+                (
+                    recoveredAccount,
+                    emailAuthMsg.proof.accountSalt,
+                    address(this)
+                )
             )
         );
-        
 
         EmailAuth guardianEmailAuth = EmailAuth(address(proxy));
-        guardianEmailAuth.updateDKIMRegistry(dkim());
-        guardianEmailAuth.updateVerifier(verifier());
+        guardianEmailAuth.initDKIMRegistry(dkim());
```

Comment:
Regarding verifier, dkimRegistry and emailAuthImpl initialization, suggestion for reducing bytecode size from @zeroknots in https://github.com/zkemail/email-recovery/pull/4/files#r1637364025

Reply:
Another small benefit is it makes the constructor more explicit for contracts inheriting from EmailAccountRecovery e.g.
```solidity
    constructor(
        address _verifier,
        address _dkimRegistry,
        address _emailAuthImpl
    )
        EmailAccountRecoveryNew(_verifier, _dkimRegistry, _emailAuthImpl)
    {
...
```

Reply:
Does it mean to replace `address public verifierAddr`, `address public dkimAddr`, and `address public emailAuthImplementationAddr` with `address internal immutable verifierAddr`, `address internal immutable dkimAddr`, and `address public emailAuthImplementationAddr` to reduce autogenerated getters?

Reply:
Yes. I think this is a minor comment for the PR in general so don't have a strong opinion on it. Completely up to you :)

Reply:
One disadvantage to making them immutable would be if the child contract wanted to use an initializer function for them. So this would restrict the use of `EmailAccountRecovery` as an implementation contract in an upgradeable flow. In this case, I think keeping them as they are would be better for developer freedom